src/nifreeze/viz/bland_altman.py

```diff
@@ -0,0 +1,284 @@
+#
+# Copyright The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Bland-Altman plot."""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.stats as stats
+
+
+def _compute_z_score(ci):
+    """Compute the z-score for the given confidence interval value.
+
+    Parameters
+    ----------
+    ci : :obj:`float`
+        Confidence interval value.
+
+    Returns
+    -------
+    :obj:`float`
+        Z-score value.
+    """
+
+    # Compute the z-score for confidence interval (two-tailed)
+    p = (1 - ci) / 2
+    q = 1 - p
+    return stats.norm.ppf(q)
+
+
+def _compute_bland_altman_features(data1, data2, ci):
+    """Compute quantities of interest for the Bland-Altman plot.
+
+    Parameters
+    ----------
+    data1 : :obj:`numpy.ndarray`
+        Data values.
+    data2 : :obj:`numpy.ndarray`
+        Data values.
+    ci : :obj:`float`
+        Confidence interval value.
+
+    Returns
+    -------
+    diff : :obj:`numpy.ndarray`
+        Differences.
+    mean_diff : :obj:`numpy.ndarray`
+        Mean differences.
+    mean : :obj:`numpy.ndarray`
+        Mean values.
+    std_diff : :obj:`numpy.ndarray`
+        Standard deviation of differences.
+    loa_lower : :obj:`numpy.ndarray`
+        Lower limit of agreement.
+    loa_upper : :obj:`numpy.ndarray`
+        Upper limit of agreement.
+    ci_mean : :obj:`numpy.ndarray`
+        Confidence interval of mean values.
+    ci_loa : :obj:`numpy.ndarray`
+        Confidence interval of limits of agreement.
+    """
+
+    # Compute mean
+    mean = np.mean([data1, data2], axis=0)
+
+    # Compute differences, mean difference, and std dev of differences
+    diff = data1 - data2
+    mean_diff = np.mean(diff)
+    std_diff = np.std(diff, axis=0)
+    # std_diff = np.std(diff, ddof=1)  # Use Bessel's correction
+
+    # Compute confidence interval limits of agreement (LoA)
+    z_score = _compute_z_score(ci)
+    loa_lower = mean_diff - z_score * std_diff
+    loa_upper = mean_diff + z_score * std_diff
+
+    n = len(diff)
+
+    # Compute t-value for the confidence intervals (CIs)
+    t_val = stats.t.ppf((1 + ci) / 2, n - 1)
+
+    # Compute confidence intervals
+
+    # Confidence interval for the mean difference
+    std_err_mean = std_diff / np.sqrt(n)
```

Comment:
Shouldn't ddof for SEM be n-1?

Reply:
Could also rely on scipy stats to compute this: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.sem.html

Reply:
`std_err_mean = stats.sem(diff, ddof=0)` would give the same result as `std_err_mean = std_diff / np.sqrt(n)`; `std_err_mean = stats.sem(diff, ddof=1)` (default) would give the same result as `std_err_mean = std_diff / np.sqrt(n-1)`.

TBH, I do not know what is the correct way in this case, as it is the case here ase well https://github.com/nipreps/nifreeze/pull/86/files#diff-f5c68a621d0e143caebb0191455696dc29900bc9316ba28f843951ec0432bdb8R89